modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -1877,12 +1874,12 @@ private CompletableFuture<Void> handleChangePendingAssignmentEvent(
                                 isRecovery
                         );
                     }), ioExecutor);
-        } else {
+        } else if (pendingAssignmentsAreForced && localMemberAssignment != null) {
             localServicesStartFuture = runAsync(() -> {
-                if (pendingAssignmentsAreForced && replicaMgr.isReplicaStarted(replicaGrpId)) {
-                    replicaMgr.resetPeers(replicaGrpId, fromAssignments(computedStableAssignments.nodes()));
-                }
+                replicaMgr.resetPeers(replicaGrpId, fromAssignments(computedStableAssignments.nodes()));
```

Comment:
BusyLock is missing.
I've added it while adding changes to the PartitionReplicaLifecycleManager. Feel free to revert if you don't agree.